TODO/type-checker-issues.md

```diff
@@ -44,45 +44,45 @@ let e: [(Double) -> String] = [
 ]
 ```
 
-All of these are completely valid Swift syntax and the expected types for every term in every expression should be obvious to any Swift programmer but in each case, the Swift type checker fails.
+上面的代码都是符合 **Swift** 语法及编程规则的，但是在编译过程中，它们都没有办法通过类型检查。
 
-## [](http://www.cocoawithlove.com/blog/2016/07/12/type-checker-issues.html#needlessly-long-compile-times)Needlessly long compile times
+## [](http://www.cocoawithlove.com/blog/2016/07/12/type-checker-issues.html#needlessly-long-compile-times)需要较长的编译时间
 
-Errors aren’t the only consequence of this problem. Try compiling the following line:
+编译报错只是 **Swift** 类型检查器缺陷带来的副作用之一，比如，你可以试试下面这个例子：
 
 ```
 let x = { String("\($0)" + "") + String("\($0)" + "") }(0)
 ```
 
-This single line does compile without error but it takes around 4 seconds to compile in Swift 2.3 and a whopping 15 seconds to compile in Swift 3 on my computer. The compiler spends almost all of this time in the Swift type checker.
+这段代码编译时不会报错，但是在我的电脑上，使用 **Swift 2.3** 将花费 **4s** 的时间，如果是使用 **Swift 3** 将会花费 **15s** 时间。编译过程中，将会花费大量的时间在类型检查上。
 
-Now, you probably don’t have many lines that take _this_ long to compile but it’s virtually guaranteed that any non-trivial Swift project is taking at least a little longer than necessarily to compile due to the same complexity problem that causes the “expression was too complex to be solved in reasonable time” error.
+现在，你可能不会遇到太多需要耗费太多时间的问题，但是在一个完整的 **Swift** 中，你将会遇到很多 `expression was too complex to be solved in reasonable time` 这样的报错信息。
 
-## [](http://www.cocoawithlove.com/blog/2016/07/12/type-checker-issues.html#unexpected-behaviors)Unexpected behaviors
+## [](http://www.cocoawithlove.com/blog/2016/07/12/type-checker-issues.html#unexpected-behaviors)不可预知的操作
 
-I want to highlight a quirk in the Swift type checker: the type checker will choose to resolve overloaded operators to non-generic overloads whenever possible. The code comments for the path that handles this specific behavior in the compiler note that this behavior exists as an optimization to avoid performance issues – the same performance issues responsible for the “expression was too complex” error.
+接下来，我将讲一点 **Swift** 类型检查器的特性：类型检查器选择尽可能的解决非泛型重载的问题。 The code comments for the path that handles this specific behavior in the compiler note that this behavior exists as an optimization to avoid performance issues – the same performance issues responsible for the “expression was too complex” error.（译者注：这里请校者一起帮忙参考下怎么处理比较好）
```

Comment:
The code comments for the path that handles this specific behavior in the compiler note that this behavior exists as an optimization to avoid performance issues – the same performance issues responsible for the “expression was too complex” error.

编译器中处理这种特定行为的路径下的代码注释对此给出了解释，这是一种避免性能问题的优化手段——即用于优化造成”语句过于复杂“报错的性能问题。